### PR TITLE
[vcpkg docs] Remove docs migrated to https://learn.microsoft.com/vcpkg and Microsoft/vcpkg-docs

### PR DESCRIPTION
**Pull request by ras0219-msft**
As part of our documentation improvements, we're consolidating the primary source of documentation onto [Microsoft Learn][1]. The docs are still 100% open source and accepting contributions at https://github.com/Microsoft/vcpkg-docs.

The https://vcpkg.io/ website will still remain as our home page with blog post links, community links, and helpful utilities like package search.

[1]: https://learn.microsoft.com/vcpkg

### CODE REVIEW

**Comment by autoantwort** [comment]
Hm I am not a big fan of removing the maintainer docs from this repo. It makes it harder to keep them in sync with the functions itself. Also I want to have them locally when working on the vcpkg repo.

**Comment by ras0219-msft** [comment]
> It makes it harder to keep them in sync with the functions itself.

While this is true, our official source of user documentation will be Microsoft/vcpkg-docs (and https://learn.microsoft.com/vcpkg) moving forward. Those locations need to be kept 100% up-to-date. We did consider several options, including leaving some docs in this repo, but ultimately decided that it would be counter-productive and confusing to have multiple "sources of truth".

To achieve that, we plan to enforce two policies:
1. Any PR that has documentation impact (such as adding a parameter to a helper function) should have a docs PR into Microsoft/vcpkg-docs opened and linked from the Microsoft/vcpkg PR before it can be merged.
2. For all new features, we will include "Added in release X-Y-Z" annotations in the documentation to ensure it is useful even for users that might not be on the latest version.

**Comment by autoantwort** [comment]
> but ultimately decided that it would be counter-productive and confusing to have multiple "sources of truth".

Ah cool so you will also shut down https://vcpkg.readthedocs.io (which will give you a lot of 404's if you click on items) and https://vcpkg.io :)  (Or instead of shutdown you will create proper redirects to [learn.microsoft.com](https://learn.microsoft.com/vcpkg))

**Comment by ras0219-msft** [comment]
New sync PR: https://github.com/microsoft/vcpkg-docs/pull/38

File: docs/maintainers/internal/vcpkg_catalog_release_process.md [hunk 1]
```diff
@@ -1,14 +0,0 @@
-# vcpkg_catalog_release_process
```
Comment (BillyONeal) [inline docs/maintainers/internal/vcpkg_catalog_release_process.md:1]: I'm assuming this file and the tool_release_process one were not migrated over; should I create these in vcpkg-tool/docs?

**Comment by BillyONeal** [comment]
> > but ultimately decided that it would be counter-productive and confusing to have multiple "sources of truth".
> 
> Ah cool so you will also shut down https://vcpkg.readthedocs.io (which will give you a lot of 404's if you click on items) and https://vcpkg.io :) (Or instead of shutdown you will create proper redirects to [learn.microsoft.com](https://learn.microsoft.com/vcpkg))

Redirects are intended.

**Comment by BillyONeal** [comment]
Resolves https://github.com/microsoft/vcpkg/issues/29019

File: docs/maintainers/internal/vcpkg_catalog_release_process.md (hunk 1 above)
Comment (ras0219-msft) [reply docs/maintainers/internal/vcpkg_catalog_release_process.md:1]: Yes, please do so if I missed it in my migration PRs.

**Comment by njakob** [comment]
Happy to see vcpkg's documentation being consolidated into a single place!

> The https://vcpkg.io/ website will still remain as our home page with blog post links, community links, and helpful utilities like package search.

Looking at the intent of the PR in its description, wasn't the goal to keep the community links on vcpkg.io? It seems they have [now been completely removed by this PR](https://github.com/microsoft/vcpkg/commit/547c030e68163025534ca3cfa372a95a88fefe24#diff-0b5ca119d2be595aa307d34512d9679e49186307ef94201e4b3dfa079aa89938L64-L66).

Was this expected @ras0219-msft?

**Pull request merged**